x-pack/solutions/observability/plugins/slo/server/lib/rules/slo_burn_rate/lib/get_group_by_object.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { set } from '@kbn/safer-lodash-set';
+
+export interface GroupByFields {
+  [x: string]: any;
+}
+
+export const unflattenObject = <T extends object = GroupByFields>(object: object): T =>
```

Comment:
There should be some jest tests for this in a `get_group_by_object.test.ts` file.